### ISSUE TITLE
king zora won't accept ruto's letter?

### ISSUE THREAD

**Issue by briaguya-ai** [briaguya-ai/rando-issue-tracker#82]
[18316.json.txt](https://github.com/briaguya-ai/rando-issue-tracker/files/9034461/18316.json.txt) reported that showing king zora ruto's letter doesn't do anything

it was also reported that showing kak guard zelda's letter wasn't doing anything

**Comment by Vsmit** [comment]
When I set the flag to move King Zora and reloaded, his text skipped to his "thanks for saving Ruto" text, even though I had never been in Jabu. I had previously gotten Zora's Sapphire from Forest Temple, so that may have had an effect, but removing it via the save editor didn't change anything.

**Comment by leggettc18** [comment]
I spent some time attempting to recreate this, but to no avail. The only time I was ever able to recreate the issue was when I had given myself Ruto's Letter via the save editor while standing on the platform, and while wearing a mask. Neither of which Vsmit could have done. He didn't pull out the save editor (as far as I am aware) until after King Zora didn't respond to the letter, and he had just sold the Skull Mask and hadn't gone back to the shop yet.

**Comment by Vsmit** [comment]
[cvars.cfg.txt](https://github.com/briaguya-ai/rando-issue-tracker/files/9035646/cvars.cfg.txt)

**Comment by briaguya-ai** [comment]
spoiler from PHIN on discord who had issues with both king zora and the guard
[22656.json.txt](https://github.com/briaguya-ai/rando-issue-tracker/files/9038056/22656.json.txt)

**Comment by briaguya-ai** [comment]
made this just about king zora because the kak guard part is probably unrelated, and also no info has been shared about it

also changing it to maybe v1 because nobody has been able to reproduce this

**Comment by leggettc18** [comment]
For what it's worth in terms of reproducing, I haven't gotten to the point of giving King Zora the letter in my run on PHIN's seed, so maybe I'll be in luck there later.

**Comment by aMannus** [comment]
Currently the option for closing Zora's Fountain is written as "Normal" to the spoiler log, but randomizer.cpp is trying to read the option as "Closed". I'm not sure if this is related.

**Comment by briaguya-ai** [comment]
> Currently the option for closing Zora's Fountain is written as "Normal" to the spoiler log, but randomizer.cpp is trying to read the option as "Closed". I'm not sure if this is related.

@aMannus where are you seeing that?

in `ParseRandomizerSettingsFile` i see

```cpp
case RSK_ZORAS_FOUNTAIN:
    if(it.value() == "Normal") {
        gSaveContext.randoSettings[index].value = 0;            
    } else if(it.value() == "Adult") {
        gSaveContext.randoSettings[index].value = 1;
    } else if(it.value() == "Open") {
        gSaveContext.randoSettings[index].value = 2;
    }
    break;
```

**Comment by briaguya-ai** [comment]
another report of this, Caio312 on disc was running build 192
[25924.json.txt](https://github.com/briaguya-ai/rando-issue-tracker/files/9058189/25924.json.txt)
[file2.sav.txt](https://github.com/briaguya-ai/rando-issue-tracker/files/9058190/file2.sav.txt)

**Comment by briaguya-ai** [comment]
spoiler from Caio312 on a more recent build (assuiming 217)
[32485.json.txt](https://github.com/briaguya-ai/rando-issue-tracker/files/9058290/32485.json.txt)

**Comment by briaguya-ai** [comment]
seems Caio312 had issues when this was in file 1, but not when it wasn't...
[file1.sav.txt](https://github.com/briaguya-ai/rando-issue-tracker/files/9059073/file1.sav.txt)

**Comment by briaguya-ai** [comment]
alright, i was able to repro by

* copying file1.sav from the comment above into my saves folder
* dragging 32485.json in to load the seed
* making a new save in file 2
* using save editor to give self bottle
* going to king z
* trying to show letter

**Comment by leggettc18** [comment]
After setting a breakpoint and stepping through z_sram.c while creating a new save file, I was able to see that the `gSaveContext.dungeonsDone` array was mirroring the same array from the first save file. I confirmed this was the case by swapping the first save file for another one that had different dungeons completed and save 2 again mirrored the first save file during creation. Additionally, I also confirmed that creating a save in slot 3 will mirror slot 2 by default.

The fix should be as simple as explicitly setting them all to 0 during save creation.
